pallets/subtensor/src/lib.rs

```diff
@@ -1204,6 +1207,13 @@ pub mod pallet {
                     // Fill stake information.
                     Owner::<T>::insert(hotkey.clone(), coldkey.clone());
 
+                    // Update Owned map
+                    let mut hotkeys = Owned::<T>::get(coldkey);
```

Comment:
can put it out of the inner loop, just update Owned once.

Reply:
Do you mean something like this ?

Reply:
```rust
let mut new_hotkeys = Vec::new();

// Inside the loop
if !OwnedHotkeys::<T>::get(coldkey).contains(hotkey) {
    new_hotkeys.push(hotkey.clone());
}

// After the loop
if !new_hotkeys.is_empty() {
    OwnedHotkeys::<T>::mutate(coldkey, |hotkeys| {
        hotkeys.extend(new_hotkeys);
    });
}
```